File: resty-generators/lib/generators/resty/setup/templates/application.rb
class Application
  include ActiveModel::Serializers::JSON
  include ActiveModel::Serializers::Xml

  attr_accessor :name, :url

  def initialize(attributes = {})
    @name = attributes['name']
    @url = attributes['url']
  end

  def attributes
    { 'name' => name, 'url' => url }
  end
end

File: resty-generators/lib/generators/resty/setup/templates/Application.java
package <%= models_package %>;

import org.codehaus.jackson.annotate.JsonCreator;
import org.codehaus.jackson.annotate.JsonProperty;

import org.fusesource.restygwt.client.Json;
import org.fusesource.restygwt.client.Json.Style;

@Json(style = Style.RAILS)
public class Application {

  private final String name;

  private final String url;

  @JsonCreator
  public Application(@JsonProperty("name") String name, 
          @JsonProperty("url") String url){
    this.url = url;
    this.name = name;
  }

  public String getName(){
    return name;
  }

  public String getUrl(){
    return url;
  }

}
